lib/go-util/join_test.go

```diff
@@ -33,3 +35,16 @@ func ExampleJoinErrsSep() {
 	// Output: test
 	// quest
 }
+
+func ExampleCamelToSnakeCase() {
+	camel := "camelCase"
+	fmt.Println(CamelToSnakeCase(camel))
+	camel = "PascalCase"
+	fmt.Println(CamelToSnakeCase(camel))
+	camel = "IPIsAnInitialismForInternetProtocol"
+	fmt.Println(CamelToSnakeCase(camel))
+
+	// Output: camel_case
+	// pascal_case
+	// ipis_an_initialism_for_internet_protocol
```

Comment:
IMO `IPIsAnInitialism` being cased to `ipis_an_initialism` is an argument for not capitalizing acronyms in camelCase or PascalCase, but our code conventions do include capitalizing them, so I can understand capitalizing it here.